### ISSUE TITLE
Solution misses references

### ISSUE THREAD

**Issue by nodermatt** [ADN-DevTech/3dsMax-Menu-Sample#1]
Hi 

I cloned the repository and opened it with Visual Studio 2017 (same as I use for usual 3ds max plugin development) but couldn't build the solution right away. At first I thought it was about missing references because the solution explorer revealed that there some problems with a couple of references; I was able to solve UiViewModel and Autodesk.Max by simply readding the files from my disk. 

What are the missing references for?
![image](https://user-images.githubusercontent.com/1467464/67636319-b9c31780-f8cf-11e9-9d84-82ad2f485276.png)

However, the problem is that the build tries to copy the assembly from the build folder to bin/assemblies. I don't know where this setting is coming from, as I don't 

![image](https://user-images.githubusercontent.com/1467464/67636449-dad83800-f8d0-11e9-8fca-56fc5800ab49.png)

Nevertheless, the actual assembly succeeds as I can just copy paste the file manually. 3ds max then loads the plugin and displays the menu:
![image](https://user-images.githubusercontent.com/1467464/67636465-09eea980-f8d1-11e9-91cd-2c84b707e759.png)

So bottom line: the project works, I am just curious about those references.
Cheers

**Comment by nodermatt** [comment]
I was able to find the missing reference in the 3ds max installation folder. I still don't know what they are for as the documentation doesn't mention them.

Source: https://cdiggins.github.io/blog/undocumented-3dsmax-dotnet-assemblies.html

**Comment by kevinvandecar** [comment]
Hi, I would encourage you to NOT refer to that blog post for reference. Undocumented assemblies are not supported and can (and have) changed. The supported assemblies are documented here: 
http://help.autodesk.com/view/3DSMAX/2020/ENU/?guid=__developer_3ds_max__net_sdk_html and listed here: http://help.autodesk.com/view/3DSMAX/2020/ENU/?guid=__developer_3ds_max__net_sdk_3ds_max__net_assemblies_html

The referenced assemblies in most "desktop" applications that Autodesk provides are wrappers. For this reason, you must specify "Copy Local" flag to be "False". The copy local flag when true (and is default when you reference a new assembly) will actually copy the assemblies to your output folder and causes all kinds of problems in the context of wrapper assemblies. When set to False, they are referenced from the "original" location, which in our case would be the 3ds Max installation folder. This way you will always have the correct ones at runtime.

The way the samples I provide are setup, is to use the 3ds Max environment variables. For example, for 3ds Max 2020 these would be: ADSK_3DSMAX_x64_2020 (this one is good for finding the .NET Assemblies) and ADSK_3DSMAX_SDK_2020 (this one is good for C++ projects to find headers and libraries). For .NET csproj files you will have to manually edit the file to adjust these as the VS IDE for editing cs projects does not accept typical "environment variables", but the build system is fine to use them. For example, in this case the project is using the 2019 variables. You can see an example of this here: https://github.com/ADN-DevTech/3dsMax-Menu-Sample/blob/9cdd6e0fc64cbe5ea460e1c81da7b75b4624c5f5/ADNMenuSample.csproj#L78
You can manually edit this csproj file to be using ADSK_3DSMAX_x64_2020 to build it against the 2020 assemblies (and you probably do not have 2019 installed, so this is why they are "missing").

**Comment by nodermatt** [comment]
@kevinvandecar Thanks for the quick response and the links to the summary page! I will refrain from consuming the undocumented assemblies; My intention was not to spread dangerous information - mea culpa.

I also found the line that was responsible for the post build copy operation:
```
    <PostBuildEvent>copy "$(TargetPath)" "$(ADSK_3DSMAX_x64_2020)bin\assemblies"
```
Changing the variable from ADSK_3DSMAX_x64_2018 to ADSK_3DSMAX_x64_2020 fixed the "error". Same goes for the referenced assemblies 👍 

Have you got an additional dotnet plugin examples apart from custom ui controls? The Lesson 7 (Utility plugin) in the learning path is a great first step but I have difficulties creating an exporter plugin (analogue to the c++ implementation).

**Comment by kevinvandecar** [comment]
Hi @nodermatt sorry, missed this before. There are samples here: https://github.com/ADN-DevTech/3dsMax-Explode-Geometry and on the blog (just look for .NET).
About writing an exporter, I would not advise it in .NET. The docs advise to not derive from plugins class. it's possible but there are some issues with virtual calls in the wrapper. The docs say this here: http://help.autodesk.com/view/3DSMAX/2020/ENU/?guid=__developer_3ds_max__net_sdk_3ds_max__net_assemblies_autodesk_max_dll_html (see the FAQ at bottom about deriving from plugins).
For an exporter, I would suggest doing it in C++. There are several samples in the SDK and docs talking about exporter topics.
If you need to work in .NET (ie. maybe support libraries are only there), then I would suggest just make it an action and assign it to UI (like menu choice) to get exporter started. Then you can export geometry and other things as you want, but it would not be part of the "export..." menu option and file types.
hope it helps,
kevin

**Comment by nodermatt** [comment]
Hi @kevinvandecar Thanks for the thorough explanation. We will proceed as you suggest.
